retrofit-converters/logansquare/src/main/java/retrofit/converter/LoganSquareConverter.java

```diff
@@ -0,0 +1,40 @@
+package retrofit.converter;
+
+import com.bluelinelabs.logansquare.LoganSquare;
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.ResponseBody;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+
+/**
+ * A {@link Converter} which uses LoganSquare for serialization and deserialization of entities.
+ */
+public class LoganSquareConverter implements Converter {
+    private static final MediaType MEDIA_TYPE = MediaType.parse("application/json; charset=UTF-8");
+
+    @Override
+    public Object fromBody(ResponseBody body, Type type) throws IOException {
+        InputStream is = body.byteStream();
+        try {
+            return LoganSquare.parse(is, (Class) type);
+        } finally {
+            try {
+                is.close();
+            } catch (IOException ignored) {
+            }
+        }
+    }
+
+    @Override
+    public RequestBody toBody(Object object, Type type) {
+        try {
+            String json = LoganSquare.serialize(object);
+            return RequestBody.create(MEDIA_TYPE, json);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Use `AssertionError` here. An `IOException` will never happen when writing to memory.